File: Programmers/level2/문자열_압축.py
def solution(s):
    if len(s) == 1:
        return 1
    result = []
    for i in range(1, (len(s) // 2) + 1):  # 가능한 반복 문자 길이
        b = ''
        cnt = 1
        tmp = s[:i]

        for j in range(i, len(s), i):  # 비교 대상
            if tmp == s[j:i + j]:
                cnt += 1
            else:
                if cnt != 1:  # 이전 반복
                    b = b + str(cnt) + tmp
                else:  # 반복 없음
                    b = b + tmp
                tmp = s[j:j + i]
                cnt = 1
        if cnt != 1:
            b = b + str(cnt) + tmp
        else:
            b = b + tmp

        result.append(len(b))
    return min(result)

File: Programmers/level2/문자열_압축/문자열_압축.java
class Solution {
    public int solution(String s) {
        int answer = s.length();

        for(int i = 1; i <= s.length()/2; i++) {
            String target = s.substring(0, i); // 시작
            String cur = ""; //현재
            int cnt = 1;

            StringBuilder sb = new StringBuilder();

            for(int start = i; start <= s.length(); start += i) {
                if (start + i >= s.length()) { // 남는 부분
                    cur = s.substring(start, s.length());
                }
                else {
                    cur = s.substring(start, start + i);
                }

                // 압축
                if(cur.equals(target)) {
                    cnt++;
                }
                // 달라서 비교 대상 변경
                else if (cnt == 1) {
                    sb.append(target);
                    target = cur;
                }
                else {
                    sb.append(cnt).append(target);
                    target = cur;
                    cnt = 1;
                }
            }
            // 남은 것
            if (i != target.length()) sb.append(target);

            answer = Math.min(answer, sb.toString().length());
        }
        return answer;
    }
}
